vagrant/guest/bin/m2setup.sh

```diff
@@ -351,7 +351,7 @@ code=
 mysql -e "use $DB_NAME" 2> /dev/null || code="$?"
 if [[ $code ]]; then
     echo "==> Creating $DB_NAME database"
-    mysql -e "create database $DB_NAME"
+    mysql -e "create database \`$DB_NAME\`"
```

Comment:
One other possibility before you merge would be to strip dashes out of the $DB_NAME